src/store/models/network.ts

```diff
@@ -943,6 +978,79 @@ const networkModel: NetworkModel = {
 
     actions.setAutoMineMode({ id, mode });
   }),
+  renameNode: thunk(
+    async (actions, { node, newName }, { getState, getStoreActions, injections }) => {
+      if (!newName) throw new Error(l('renameErr', { newName }));
+
+      if (node.status === Status.Started) {
+        await actions.stop(node.networkId);
+      }
+
+      const networks = getState().networks;
+      const network = networks.find(n => n.id === node.networkId);
+      if (!network) throw new Error(l('networkByIdErr', { networkId: node.networkId }));
+
+      let updatedNode;
+      switch (node.type) {
+        case 'lightning':
+          updatedNode = network?.nodes.lightning.find(n => n.id === node.id) as Node;
+          getStoreActions().designer.renameLightningNode({
+            name: newName,
+            nodeId: updatedNode.name,
+            backendName: updatedNode.backendName,
+          });
+          updatedNode.name = newName;
+          updatedNode.paths = getLndFilePaths(newName, network);
+          break;
+        case 'bitcoin':
+          updatedNode = network?.nodes.bitcoin.find(n => n.id === node.id) as BitcoinNode;
+          getStoreActions().designer.renameBitcoinNode({
+            name: newName,
+            nodeId: updatedNode.name,
+          });
+          updatedNode.name = newName;
+          // TODO: Update the path of the bitcoin node
+          break;
+        case 'tap':
+          updatedNode = network?.nodes.tap.find(n => n.id === node.id) as Node;
+          getStoreActions().designer.renameTapNode({
+            name: newName,
+            nodeId: updatedNode.name,
+            lndName: updatedNode.lndName,
+          });
+          updatedNode.name = newName;
+          updatedNode.paths = getTapdFilePaths(newName, network);
+          break;
+        default:
+          throw new Error('Invalid node type');
+      }
+
+      // Rename the docker volume data from disk
+      const volumeDir = node.implementation.toLocaleLowerCase().replace('-', '');
+      const oldPath = join(network.path, 'volumes', volumeDir, node.name);
+      const newPath = join(network.path, 'volumes', volumeDir, newName);
+
+      fs.rename(oldPath, newPath)
+        .then(() => console.log(`Renamed directory from ${oldPath} to ${newPath}`))
+        .catch((err: any) => console.error(`Error renaming directory: ${err}`));
```

Comment:
The renaming of dirs should be done in the `dockerService` since that is also responsible for creating them. It can also handle saving the compose file as well.